hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceImpl.java

```diff
@@ -84,12 +93,15 @@ public void registerSchemas(@NonNull final SchemaRegistry registry) {
         registry.register(new V0500ContractSchema());
     }
 
-    /**
-     * Create the metrics for the smart contracts service. This needs to be delayed until _after_
-     * the metrics are available - which happens after `Hedera.initializeStatesApi`.
-     */
-    public void registerMetrics() {
-        component.contractMetrics().createContractMetrics();
+    @Override
+    public void onStateInitializedForGenesis() {
+        // Force call translators to be instantiated now, so that all the system contract methods
+        // will be registered, so the secondary metrics can be created.  (Left to its own devices
+        // Dagger would delay instantiating them until transactions started flowing.)
+        final var allTranslators = allCallTranslators();
```

Comment:
Is there a reason for the `allTranslators` variable?

Reply:
Comment didn't explain it properly?  Has to do with Dagger being extremely lazy.  Normally the translators aren't instantiated until transactions begin to flow, because that's when they're first needed, thus first created.  But I want to create the metrics _before_ the transactions begin to flow, so as not to slow down actual operation of the node.